File: 1009-complement-of-base-10-integer/1009-complement-of-base-10-integer.java
class Solution {
    public int bitwiseComplement(int n) {
        if (n == 0) return 1;
        
        int i = 31;
        while (i >= 0 && (n & (1 << i)) == 0) {
            i--;
        }
        for (; i >= 0; i--) {
            n = n ^ (1 << i);
        }
        return n;
    }
}

File: LeetCode/1009-complement-of-base-10-integer/1009-complement-of-base-10-integer.py
class Solution:
    def bitwiseComplement(self, n: int) -> int:
        if n == 0:
            return 1
        
        i = 31
        while i >= 0 and (n & (1 << i)) == 0:
            i -= 1
        
        while i >= 0:
            n = n ^ (1 << i)
            i -= 1
            
        return n